src/Framework/AspNetCoreAnalyzers/test/RouteEmbeddedLanguage/RoutePatternAnalyzerTests.cs

```diff
@@ -555,4 +561,33 @@ static void Main()
         // Assert 2
         Assert.Empty(diagnostics2);
     }
+
+    [Fact]
+    public async Task ConcatString_DetectLanguage_WarningsReported()
+    {
+        // Arrange
+        var builder = new StringBuilder();
+        builder.AppendLine("""
+    class Program
+    {
+        static void Main() { }
+        // lang=Route
+        static readonly string _s =
+    """);
+        for (var i = 0; i < 100; i++)
```

Comment:
2000 or more?

Reply:
The intent of the test is deeply nested concat strings are all classified as routes. 100 is enough.

Reply:
I changed it to 2000 and found another example of `HasLanguageComment` being repeatedly called. Fixed, and I left this at 2000.